renku/core/commands/init.py

```diff
@@ -0,0 +1,148 @@
+    """Fetch the template and checkout the relevant data.
+
+    :param source: url or full path of the templates repository
+    :param ref: reference for git checkout - branch, commit or tag
+    :param tempdir: temporary work directory path
+    :return: template manifest Path
+    """
+    if tempdir is None:
+        tempdir = Path(tempfile.mkdtemp())
+
+    # clone the repo locally without checking out files
+    template_repo = git.Repo.clone_from(source, tempdir, no_checkout=True)
+
+    # fetch ref and set the HEAD
+    template_repo.remotes.origin.fetch(ref)
+    try:
+        template_repo.head.reset(template_repo.commit(ref))
+    except git.exc.BadName:
+        ref = 'origin/{0}'.format(ref)
+        template_repo.head.reset(template_repo.commit(ref))
+    git_repo = git.Git(tempdir)
+
+    # checkout the manifest
+    git_repo.checkout(TEMPLATE_MANIFEST)
+    return tempdir / TEMPLATE_MANIFEST
+
+
+def validate_template(template_path):
+    """Validate a local template.
+
+    :param template_path: path of the template to validate
+    """
+    # TODO: implement a better check
+    required_folders = [RENKU_HOME]
+    required_files = ['{0}/metadata.yml'.format(RENKU_HOME), 'Dockerfile']
+    for folder in required_folders:
+        if not Path(template_path, folder).is_dir():
+            raise ValueError(
+                'Folder {0} is required for the template to be valid'.
+                format(folder)
+            )
+    for file in required_files:
+        if not Path(template_path, file).is_file():
+            raise ValueError(
+                'File {0} is required for the template to be valid'.
+                format(file)
+            )
+    return True
+
+
+def validate_template_manifest(manifest):
+    """Validate manifet content.
+
+    :param manifest: manifest file content
+    """
+    if not isinstance(manifest, list):
+        raise ValueError((
+            'The repository doesn\'t contain a valid',
+            '"{0}" file'.format(TEMPLATE_MANIFEST)
+        ))
+    for template in manifest:
+        if not template['name']:
+            raise ValueError((
+                'Every template listed in "{0}"',
+                ' must have a name'.format(TEMPLATE_MANIFEST)
+            ))
+        for attribute in ['folder', 'description']:
+            if not template[attribute]:
+                raise ValueError((
+                    'Template "{0}" doesn\'t have a {1} attribute'.format(
+                        template['name'], attribute
+                    )
+                ))
+    return True
+
+
+def read_template_manifest(folder, checkout=False):
+    """Extract template metadata from the manifest file.
+
+    :param folder: path where to find the template manifest file
+    :param checkout: checkout the template folder from local repo
+    """
+    manifest_path = folder / TEMPLATE_MANIFEST
+
+    with manifest_path.open('r') as fp:
+        manifest = yaml.safe_load(fp)
```

Comment:
If you don't read and write transactionally then you probably don't need file handling via contextmanagers. You could achieve the same with 

```
manifest = yaml.safe_load(manifest_path.read_text())
```
```